src/idl_parser.cpp

```diff
@@ -914,8 +914,20 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
   size_t fieldn = 0;
   auto err = ParseTableDelimiters(fieldn, &struct_def,
                                   [&](const std::string &name) -> CheckedError {
-    auto field = struct_def.fields.Lookup(name);
-    if (!field) {
+  const char *schema_key = "$schema";
+  bool is_schema_key = true;
+  for (std::string::size_type i = 0; i < name.size(); ++i) {
```

Comment:
lets not hand-code our own string-compares. Do people actually use `$SCHEMA` or `$Schema` ? If not, do a straight string compare. If all 3 exist, I'd say enter all 3 into a set<string> auto_skip_fields created in the parser at initialization, and test against that.